extensions/narayana-stm/deployment/src/main/java/io/quarkus/narayana/stm/deployment/NarayanaSTMProcessor.java

```diff
@@ -39,14 +45,30 @@
 
     // register classes in need of reflection
     @BuildStep
-    ReflectiveClassBuildItem register(BuildProducer<FeatureBuildItem> feature) {
+    ReflectiveClassBuildItem registerFeature(BuildProducer<FeatureBuildItem> feature) {
         feature.produce(new FeatureBuildItem(FeatureBuildItem.NARAYANA_STM));
 
         return new ReflectiveClassBuildItem(true, false,
+                ShadowNoFileLockStore.class.getName(),
                 CheckedActionFactoryImple.class.getName(),
                 Lock.class.getName());
     }
 
+    // the software transactional memory implementation does not require a TSM
+    // so disable it at native image build time
+    @BuildStep()
+    public NativeImageSystemPropertyBuildItem substrateSystemPropertyBuildItem() {
+        return new NativeImageSystemPropertyBuildItem("CoordinatorEnvironmentBean.transactionStatusManagerEnable", "false");
+    }
+
+    // the software transactional memory implementation does not require a TSM
+    // so disable it at runtime
+    @BuildStep()
+    @Record(RUNTIME_INIT)
+    public void configureRuntimeProperties(NarayanaSTMRecorder recorder) {
+        recorder.disableTransactionStatusManager();
```

Comment:
Just for my own curiosity: Why do we have this? Isn't 
```
return new NativeImageSystemPropertyBuildItem("CoordinatorEnvironmentBean.transactionStatusManagerEnable", "false");
``` 
done above enough?

Reply:
@machi1990 I will retest the extension without it ...

Reply:
It is required since when we run the non native image (standard java jar) we do not want the TSM to be started.

Reply:
Ah sure. Thanks  for the explanation.